benchmark-service/benchmark/submit.py

```diff
@@ -0,0 +1,257 @@
+
+    @staticmethod
+    def from_short_str(s):
+        pieces = s.split("/")
+        assert len(pieces) == 2, f'{pieces} {s}'
+        return Repo(pieces[0], pieces[1])
+
+    def short_str(self):
+        return f'{self.owner}/{self.name}'
+
+
+class Commit:
+    def __init__(self, repo: Repo, sha: str):
+        self.repo = repo
+        self.sha = sha
+
+    @staticmethod
+    def from_str(s):
+        commit = GITHUB_COMMIT_REGEX.fullmatch(s).groupdict()
+        assert commit
+        repo = Repo(commit['owner'], commit['name'])
+        sha = commit['sha']
+        return Commit(repo, sha)
+
+    def short_str(self):
+        return f'{self.repo.owner}/{self.repo.name}:{self.sha}'
+
+    def __str__(self):
+        return self.short_str()
+
+    def repo_dir(self):
+        return f'/tmp/repo/{self.repo.owner}/{self.repo.name}'
+
+    def checkout_script(self):
+        return f'''
+mkdir -p { shq(self.repo_dir()) }
+if [ ! -d { shq(self.repo_dir()) }/.git ]
+then
+    cd /
+    rm -rf { shq(self.repo_dir()) }
+    git clone { shq(self.repo.url) } { shq(self.repo_dir()) }
+    cd { shq(self.repo_dir()) }
+else
+    cd { shq(self.repo_dir()) }
+    git reset --hard
+    git fetch -q origin
+fi
+git checkout {self.sha}
+'''
+
+
+def submit(hail_code: Commit,
+           benchmark_code: Commit,
+           test_names: Set[str],
+           n_replicates: int,
+           n_iters: int):
+
+    sync_check_shell(benchmark_code.checkout_script())
+
+    sys.path.insert(0, f'{benchmark_code.repo_dir()}/benchmark/python/benchmark_hail')
+
+    importlib.invalidate_caches
+    from benchmark_hail.run.resources import all_resources
+    from benchmark_hail.run.utils import list_benchmarks
+
+    output_file = f'gs://hail-benchmarks-2/benchmark/{hail_code.sha}-{benchmark_code.sha}.json'
+
+    b = hb.Batch(name=f'benchmark-{hail_code.sha}',
+                 backend=hb.ServiceBackend(billing_project='hail'),
+                 default_image=BENCHMARK_IMAGE,
+                 default_cpu='2',
+                 attributes={'output_file': output_file,
+                             'n_replicates': str(n_replicates),
+                             'n_iters': str(n_iters),
+                             'image': str(BENCHMARK_IMAGE),
+                             'hail_code': str(hail_code),
+                             'benchmark_code': str(benchmark_code)})
+
+    build_hail = b.new_job('build_hail_wheel')
+    build_hail.command(f'''
+ set -ex
+ { hail_code.checkout_script() }
+ cd hail
+ time ./gradlew --version
+ time make jars python-version-info wheel
+ time (cd python && zip -r hail.zip hail hailtop)
+ time tar czf test.tar.gz -C python test
+ time tar czf resources.tar.gz -C src/test resources
+ time tar czf data.tar.gz -C python/hail/docs data
+ (cd build/deploy/dist/ && tar -cvf wheel-container.tar hail-*-py3-none-any.whl)
+ cp build/deploy/dist/hail-*-py3-none-any.whl {build_hail.wheel}
+ cp build/libs/hail-all-spark.jar {build_hail.jars}
```

Comment:
I'm not sure if you need the jar file. I think you only reference the wheel file?